### ISSUE TITLE
AssertionError when loading example model

### ISSUE THREAD

**Issue by keithnewman** [pharmpy/pharmpy#1872]
### Bug report

When attempting to load the "pheno" or "pheno-linear" example models in pharmpy, using the [example in the documentation](https://github.com/pharmpy/pharmpy/blob/3033e652b74f69cb0e9dcf9f026d4c3b6951bce2/src/pharmpy/modeling/common.py#L344-L345) for the method, an assertion error occurs. The same error message appears when attempting to load some of my own model files.

```
$ pip3 install pharmpy-core
$ python3
Python 3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pharmpy
>>> pharmpy.__version__
'0.97.0'
>>> from pharmpy.modeling import load_example_model
>>> mod=load_example_model("pheno")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/[censored_path]/lib/python3.10/site-packages/pharmpy/modeling/common.py", line 381, in load_example_model
    model = read_model(path)
  File "/[censored_path]/lib/python3.10/site-packages/pharmpy/modeling/common.py", line 51, in read_model
    model = Model.parse_model(path)
  File "/[censored_path]/lib/python3.10/site-packages/pharmpy/model/model.py", line 640, in parse_model
    model = model_module.parse_model(code, path)
  File "/[censored_path]/lib/python3.10/site-packages/pharmpy/model/external/nonmem/model.py", line 313, in parse_model
    control_stream = parser.parse(code)
  File "/[censored_path]/lib/python3.10/site-packages/pharmpy/model/external/nonmem/nmtran_parser.py", line 46, in parse
    record = create_record(separator + s)
  File "/[censored_path]/lib/python3.10/site-packages/pharmpy/model/external/nonmem/records/factory.py", line 93, in create_record
    root = record_parser_class(content).root
  File "/[censored_path]/lib/python3.10/site-packages/pharmpy/internals/parse/generic.py", line 383, in __init__
    self.root = self.parse(buf)
  File "/[censored_path]/lib/python3.10/site-packages/pharmpy/internals/parse/generic.py", line 403, in parse
    root = processor(self.buffer, root)
  File "/[censored_path]/lib/python3.10/site-packages/pharmpy/internals/parse/ignored.py", line 109, in with_ignored_tokens
    + list(_tokenize_ignored_characters(source, new_tree.meta.end_pos, len(source)))
  File "/[censored_path]/lib/python3.10/site-packages/pharmpy/internals/parse/ignored.py", line 38, in _tokenize_ignored_characters
    assert first == '&'
AssertionError
```

Looking at the commit history, this appears to line causing the issue appears to have been created in https://github.com/pharmpy/pharmpy/commit/b9846f46adcc75c7134c06004e78b5745a7cfbcd of v0.84.1

When running on v0.84.0, the example dataset loads fine, which is the behaviour I was expecting.

```
$ pip3 install pharmpy-core==0.84.0
$ python3
Python 3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pharmpy.modeling import load_example_model
>>> mod = load_example_model("pheno")
>>> mod
<Pharmpy model object pheno>          
>>> mod.dataset
     ID   TIME   AMT  WGT  APGR    DV  FA1  FA2
0     1    0.0  25.0  1.4   7.0   0.0  1.0  1.0
1     1    2.0   0.0  1.4   7.0  17.3  0.0  0.0
[...snip...]
743  59  146.8   0.0  1.1   6.0  40.2  0.0  0.0

[744 rows x 8 columns]
>>> import pharmpy
>>> pharmpy.__version__
'0.84.0'
```

**Comment by stellabelin** [comment]
Hi, thanks for reporting this! I could not recreate the error locally (with 0.97.0), I'm thinking if it has something to do with the version of any of the dependencies? Which version of e.g. lark are you using?

**Comment by stellabelin** [comment]
Hi again! I had a colleague that encountered the same bug, it has to do with the latest version of lark, I'll see if I can make a quick fix, in the meantime what you can do is downgrade lark to 1.1.5.

**Comment by keithnewman** [comment]
Thanks @stellabelin 

I was using lark 1.1.6

```
>>> import lark
>>> lark.__version__
'1.1.6'
```

And having installed lark 1.1.5, it does work now, thanks.

```
$ python3
Python 3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import lark
>>> lark.__version__
'1.1.5'
>>> import pharmpy
>>> pharmpy.__version__
'0.97.0'
>>> from pharmpy.modeling import load_example_model
>>> mod=load_example_model("pheno")
>>> mod
<Pharmpy model object pheno>
>>> mod.dataset
     ID   TIME   AMT  WGT  APGR    DV  FA1  FA2
0     1    0.0  25.0  1.4   7.0   0.0  1.0  1.0
1     1    2.0   0.0  1.4   7.0  17.3  0.0  0.0
[...snip...]
743  59  146.8   0.0  1.1   6.0  40.2  0.0  0.0

[744 rows x 8 columns]
```

**Comment by stellabelin** [comment]
Thanks again for reporting this, it was a bug in lark 1.1.6 which has been fixed in lark 1.1.7, you should be able to update to the latest lark now. We also have a fix in Pharmpy for lark 1.1.6. Closing the issue.
